hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql

```
@@ -19,7 +19,7 @@ create table if not exists address_book_service_endpoint
     consensus_timestamp bigint      not null,
     ip_address_v4       varchar(15) not null,
     node_id             bigint      not null,
-    port                integer     not null
+    port                integer     default 0 not null
```

I'm still not a fan of 0 for default as it's a valid port. If we're defaulting to valid ports then we should default to 50211. If we're defaulting to invalid ports then it should be -1.

---

I agree, I lean towards the -1.

---

Also in circling back there's a slight difference. The original data in both cases was indeed 0. However, we're dealing with trying to differentiate between historic cases where there was a valid IP and we took 0 and set it to null and are now setting it to -1 vs when we originally didn't modify the port value.

---

Updated